dts/bindings/adc/st,stm32-adc.yaml

```diff
@@ -21,6 +21,22 @@ properties:
     "#io-channel-cells":
       const: 1
 
+    st,adc-prescaler:
+      required: false
+      type: array
+      description: |
+        The ADC Clock specifier at the input of the peripheral.
+        On some series, it could be either ASYNChronous (derived from the domain clock)
+        or SYNChronous (derived from the AHB or APB).
+        In each case, a specific prescaler can be applied to reduce the input frequency.
+        It is combined to the st,prescaler to define the exact ADC clock freq.
+        Refer to the RefMan to know the possible schemes.
+        The property must have two integer elements:
+        - first:  SYNC or ASYNC to select synchronous or asynchronous specific clock
+        - second: divider which is an integer from 1 to 256
+        (some stm32 may have only limited nb of divider. For example on stm32F2 :
+        SYNC of 4 (possibly 2, 4, 6, 8) will give st,prescaler = <SYNC 4>;
```

Comment:
It feels weird to mix data types like this in an array.  Feels like there should be:
`st,adc-prescale-type' (SYNC/ASYNC) and `st-adc-prescaler`.

Reply:
Is that blocking? 
FWIW, this is what we're doing already for pinctrl:
`pinmux = <STM32_PINMUX('A', 3, ANALOG)>;`

We'll need to add much more props in these ADC bindings in the end, so if there's a way to save a few, I would be pleased. Anyway, up to you!

Reply:
Giving two properties (st,adc-prescaler-type' and 'st,adc-prescaler') was a (much more simpler) option that was not retained to combine both information in a single property.

Reply:
@galak Is that bocking ?

Reply:
@galak ping. Also note that one of the reason for combining these props is that both registers are strictly linked:
`he ADC prescaler value is applied only when CKMODE[1:0] = 0b00.` and they combine in a single parameter of the function `LL_ADC_SetCommonClock`.
So it's possible to split in 2 properties, but I do think current proposal makes things easier for the user.

Reply:
@mbolivar-nordic thoughts ?

Reply:
> @mbolivar-nordic thoughts ?

No thoughts from me -- I'm not maintaining the individual bindings anymore, so you'll have to get a reply from @galak. 72637ac746a7fe92bc2d76f21e8c4a53eceb17bd